paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -190,6 +191,10 @@ void ListenAndServOp::RunAsyncLoop(framework::Executor *executor,
     block_list.push_back(blkid);
   }
   auto optimize_prepared = executor->Prepare(*program, block_list);
+  // execute global block if needed
+  if (block_list[0] == 1 && id_to_grad.count(1) == 0) {
```

Comment:
```cpp
if (block_list[0] == 1 && id_to_grad.count(0) == 0)
```
?

Reply:
Block(idx=0) would be `listen_and_serv` block, Block(idx=1) would be global block or not.

Reply:
I think Block(idx=1).parent_idx always be -1.